Panic (SIGSEGV) when generating SARIF output with v2.13.0

### Summary 
I'm facing panic on SARIF generation step.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6bebf4]

goroutine 1 [running]:
github.com/securego/gosec/v2/report/sarif.GenerateReport({0xc001bdd6d0, 0x1, 0x1}, 0xc001893300)
	/build/report/sarif/formatter.go:30 +0x254
github.com/securego/gosec/v2/report/sarif.WriteReport({0x865320, 0xc001589710}, 0x3?, {0xc001bdd6d0?, 0xb?, 0x1?})
	/build/report/sarif/writer.go:12 +0x3b
github.com/securego/gosec/v2/report.CreateReport({0x865320, 0xc001589710}, {0x7ffefcd3a238, 0x7f199b0c5fb8?}, 0x30?, {0xc001bdd6d0?, 0x40?, 0xc0028a0000?}, 0xc001893300)
	/build/report/formatter.go:77 +0x245
main.saveReport({0x7ffefcd3a243?, 0x1?}, {0x7ffefcd3a238, 0x5}, {0xc001bdd6d0, 0x1, 0x1}, 0x203000?)
	/build/cmd/gosec/main.go:261 +0xf2
main.main()
	/build/cmd/gosec/main.go:457 +0x10b7
```
The part of log has been taken from [here](https://github.com/wavesplatform/gowaves/runs/7920735004?check_suite_focus=true).

### Steps to reproduce the behavior
Run `gosec -no-fail -fmt sarif -out gosec.sarif ./...` against [wavesplatform/gowaves](https://github.com/wavesplatform/gowaves/tree/e1a20a6cd2ee8fe81aa9e9f7a003a9030a5f4f81) project.

### gosec version

`2.13.0`

### Go version (output of 'go version')

`go version go1.19 darwin/amd64`

### Operating system / Environment

 macOS 12.5.1 (Kernel Version: Darwin 21.6.0) and github runners

### Expected behavior
Works without panics

### Actual behavior
Panics

--- a/cwe/types.go
+++ b/cwe/types.go
@@ -19,7 +19,11 @@ func (w *Weakness) SprintURL() string {
 
 // SprintID format the CWE ID
 func (w *Weakness) SprintID() string {
-	return fmt.Sprintf("%s-%s", Acronym, w.ID)
+	id := "0000"
+	if w != nil {
+		id = w.ID
+	}
+	return fmt.Sprintf("%s-%s", Acronym, id)
 }
 
 // MarshalJSON print only id and URL

--- a/report/golint/writer.go
+++ b/report/golint/writer.go
@@ -15,7 +15,7 @@ func WriteReport(w io.Writer, data *gosec.ReportInfo) error {
 
 	for _, issue := range data.Issues {
 		what := issue.What
-		if issue.Cwe.ID != "" {
+		if issue.Cwe != nil && issue.Cwe.ID != "" {
 			what = fmt.Sprintf("[%s] %s", issue.Cwe.SprintID(), issue.What)
 		}
 

--- a/report/junit/formatter.go
+++ b/report/junit/formatter.go
@@ -8,11 +8,15 @@ import (
 )
 
 func generatePlaintext(issue *gosec.Issue) string {
+	cweID := "CWE"
+	if issue.Cwe != nil {
+		cweID = issue.Cwe.ID
+	}
 	return "Results:\n" +
 		"[" + issue.File + ":" + issue.Line + "] - " +
 		issue.What + " (Confidence: " + strconv.Itoa(int(issue.Confidence)) +
 		", Severity: " + strconv.Itoa(int(issue.Severity)) +
-		", CWE: " + issue.Cwe.ID + ")\n" + "> " + html.EscapeString(issue.Code)
+		", CWE: " + cweID + ")\n" + "> " + html.EscapeString(issue.Code)
 }
 
 // GenerateReport Convert a gosec report to a JUnit Report

--- a/report/sarif/formatter.go
+++ b/report/sarif/formatter.go
@@ -27,12 +27,14 @@ func GenerateReport(rootPaths []string, data *gosec.ReportInfo) (*Report, error)
 	weaknesses := make(map[string]*cwe.Weakness)
 
 	for _, issue := range data.Issues {
-		_, ok := weaknesses[issue.Cwe.ID]
-		if !ok {
-			weakness := cwe.Get(issue.Cwe.ID)
-			weaknesses[issue.Cwe.ID] = weakness
-			cweTaxon := parseSarifTaxon(weakness)
-			cweTaxa = append(cweTaxa, cweTaxon)
+		if issue.Cwe != nil {
+			_, ok := weaknesses[issue.Cwe.ID]
+			if !ok {
+				weakness := cwe.Get(issue.Cwe.ID)
+				weaknesses[issue.Cwe.ID] = weakness
+				cweTaxon := parseSarifTaxon(weakness)
+				cweTaxa = append(cweTaxa, cweTaxon)
+			}
 		}
 
 		r, ok := rulesIndices[issue.RuleID]
@@ -97,6 +99,9 @@ func parseSarifRule(issue *gosec.Issue) *ReportingDescriptor {
 }
 
 func buildSarifReportingDescriptorRelationship(weakness *cwe.Weakness) *ReportingDescriptorRelationship {
+	if weakness == nil {
+		return nil
+	}
 	return &ReportingDescriptorRelationship{
 		Target: &ReportingDescriptorReference{
 			ID:            weakness.ID,

